### ISSUE TITLE
Regression - slider handle moves only a tiny bit and stops

### ISSUE THREAD

**Issue by fidian** [angular-slider/ngx-slider#111]
I experienced no issues with 1.2.1. Today we upgraded to 1.2.3 and found that the slider handle only moves a few pixels and then stops moving. The mousemove event handler appears to not be fired.

We're using Angular 8.0.1 and Chrome.

I don't think I can spend the time at work to make a simple reproduction of this app, but maybe you could tell me some places to add debugging statements and I'd be happy to assist with troubleshooting.

**Comment by piotrdz** [comment]
Can you provide the slider options that you are using?

**Comment by fidian** [comment]
I am using the slider to select a range.

```
<ng5-slider
    class="D(f) Fxg(1) Fxs(1)"
    [value]="value"
    (valueChange)="valueChange.emit($event)"
    [highValue]="highValue"
    (highValueChange)="highValueChange.emit($event)"
    [options]="options"
></ng5-slider>
```

The options object looks like this.

```
{
    floor: this.min,
    ceil: this.max,
    hideLimitLabels: true,
    hidePointerLabels: true,
    draggableRange: true
}
```

The min and max are set to 35 and 80, respectively.

Do you suspect it's one of those settings?

**Comment by piotrdz** [comment]
I have tried to reproduce this with the exact options you gave on Angular 8 with various browsers (Chrome, Firefox, Edge) but without much luck. I didn't see anything wrong when it comes to mouse event handling.

I have found one thing to do with slider initialisation, which I just fixed in v1.2.4, but this is likely unrelated.

However, given that you are using `valueChange` and `highValueChange`, I suspect the problem is with your bindings here. In v1.2.0 I rewrote majority of code handling input/output bindings, while also fixing a number of issues regarding emitting output changes in `valueChange`/`highValueChange`.

It depends on what code you have subscribed to `valueChange`/`highValueChange`, but if this code is doing some intensive computation, it can slow down the slider. Consider for example [this Stackblitz](https://stackblitz.com/edit/ng5-slider-issue-111-possible-reproduction) that illustrates it.

**Comment by gentle9** [comment]
I am having the same problem.
After updating from 1.2.1 to 1.2.4 the slider doesnt move with user interaction. 
`<ng5-slider [options]="rangeSliderOptions" formControlName="slider"></ng5-slider>`
`    rangeSliderOptions: Options = {
        floor: -12,
        ceil: 12,
        step: 1,
        draggableRange: true
    };`

**Comment by fidian** [comment]
You're right that it has to do with `valueChange` and `highValueChange`, or something that those events trigger. I disabled `valueChange` and now the left-hand slider works as expected and the right-hand one is busted. It doesn't operate the same as your Stackblitz. Regardless how long you wait, you can't move it again. It moves a single jump and that's it.

Here's the element with `valueChange` disabled and `highValueChange` left intact.

![captured](https://user-images.githubusercontent.com/428348/60101240-0380d300-9721-11e9-8491-e3602d283ac7.gif)

I will see if I can diagnose this issue further.

**Comment by fidian** [comment]
It happens when the options object is replaced with a new object. This [Stackblitz](https://stackblitz.com/edit/ng5-slider-issue-111) reproduces the issue.

**Comment by piotrdz** [comment]
Ah, I did not expect that. I see what is happening now.

When the slider options are changed, the slider resets itself, including unbinding and rebinding event handlers. I am actually surprised that this was working before.

I will change it so that the events are unbound and rebound only when a certain subset of options change - the subset relevant to how the events are being handled. Otherwise, the events will be left alone, which should enable the previous functionality.

**Comment by lixaotec** [comment]
any workaround?

I´m facing the same issues

**Comment by KyselMatus** [comment]
I managed to get around this by assigning the desired properties of 'options' object instead of passing whole new options.

Do this:

`
options.floor = this.min;
options.ceil = this.max;
options.hideLimitsLabel = true;
`

instead of this:

`
options = {
    floor: this.min,
    ceil: this.max,
    hideLimitsLabel: true
}
`

**Comment by piotrdz** [comment]
This is now fixed in master, pending next release.

**Comment by piotrdz** [comment]
Released in v2.0.1. Closing.
